major build updates 2023-01

brody2consult opened the pull request:
__UPDATED:__

supersedes PR #43; ~~includes PR #42~~

~~TODO:~~

- ~~update targetSdkVersion -> 33 - I had some trouble with this, may need some investigation~~ _(looks like this was already done in master)_

---

__STATUS:__

~~I would favor keeping this as a draft for now. Better if we can test without the SDK 31 updates, not everyone may be ready for this yet.~~

_Now that this project targets recent SDK 33, I would favor this over the minor updates proposed in PR #43_

brodybits commented:
Looks like e113fa26f206a78a2780a46d555686b99b617a08 would be overridden by 2cf379d60579c619e61e9923ee08a5c106a2055a on master ... I would like to rebase the other changes

Commit:
update Gradle wrapper -> 7.6

./gradlew wrapper --gradle-version 7.6

- update wrapper properties
- update wrapper JAR
- update wrapper scripts

ref:

- https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper
- https://gradle.org/releases/

Commit:
update com.android.tools.build:gradle -> 7.3.1

ref:

- https://mvnrepository.com/artifact/com.android.tools.build/gradle?repo=google

Commit:
update minSdkVersion -> 21

Commit:
update androidx.sqlite:sqlite -> 2.2.0

(requires minimum Android SDK 31)

ref:

- https://mvnrepository.com/artifact/androidx.sqlite/sqlite
